src/signOpCert.c

```diff
@@ -0,0 +1,169 @@
+#include "common.h"
+
+#include "signOpCert.h"
+#include "keyDerivation.h"
+#include "endian.h"
+#include "state.h"
+#include "uiHelpers.h"
+#include "uiScreens.h"
+#include "securityPolicy.h"
+#include "messageSigning.h"
+
+static ins_sign_op_cert_context_t* ctx = &(instructionState.signOpCertContext);
+
+
+static int16_t RESPONSE_READY_MAGIC = 31678;
+
+// forward declaration
+static void signOpCert_ui_runStep();
+enum {
+	UI_STEP_WARNING = 100,
+	UI_STEP_CONFIRM_START,
+	UI_STEP_DISPLAY_POOL_COLD_KEY_PATH,
+	UI_STEP_DISPLAY_KES_PUBLIC_KEY,
+	UI_STEP_DISPLAY_KES_PERIOD,
+	UI_STEP_DISPLAY_ISSUE_COUNTER,
+	UI_STEP_CONFIRM,
+	UI_STEP_RESPOND,
+	UI_STEP_INVALID,
+};
+
+void signOpCert_handleAPDU(
+        uint8_t p1,
+        uint8_t p2,
+        uint8_t *wireDataBuffer,
+        size_t wireDataSize,
+        bool isNewCall
+)
+{
+	// Initialize state
+	if (isNewCall) {
+		explicit_bzero(ctx, SIZEOF(*ctx));
+	}
+	ctx->responseReadyMagic = 0;
+
+	// Validate params
+	VALIDATE(p1 == P1_UNUSED, ERR_INVALID_REQUEST_PARAMETERS);
+	VALIDATE(p2 == P2_UNUSED, ERR_INVALID_REQUEST_PARAMETERS);
+
+	{
+		TRACE_BUFFER(wireDataBuffer, wireDataSize);
```

Comment:
For fixed-sized stuff we usually use the "wireheader" version: I think the reason to use it is because it nicely lists all the required things in a single place.
see e.g. signTx_handleInitAPDU

Reply:
cold key path is a variable length entry - in theory I can have everything but the path in such struct and deal with the path separately but I'm not sure if it's worth it - what do you think?